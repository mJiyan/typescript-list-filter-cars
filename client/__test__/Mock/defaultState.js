const defaultState = {
    make: 'Ford',
    makes: [
        "ALFA ROMEO",
        "AUDI",
        "AUSTIN",
        "BARKAS",
        "BMW",
        "CADILLAC",
        "CHEVROLET",
        "CHRYSLER",
        "CITROEN",
        "CORVETTE",
        "DACIA",
        "DAEWOO",
        "DAF",
        "DAIHATSU",
        "DODGE",
        "FIAT",
        "FORD",
        "HDPIC",
        "HONDA",
        "HYUNDAI",
        "INFINITI",
        "ISUZU",
        "IVECO",
        "JAGUAR",
        "JEEP",
        "KIA",
        "LADA",
        "LANCIA",
        "LAND ROVER",
        "LANDWIND",
        "LDV",
        "LEXUS",
        "MAZDA",
        "MERCEDES-BENZ",
        "MG",
        "MINI (BMW)",
        "MINI (ROVER)",
        "MITSUBISHI",
        "NISSAN",
        "OPEL",
        "PEUGEOT",
        "PONTIAC",
        "PORSCHE",
        "PROTON",
        "RENAULT",
        "ROVER",
        "SAAB",
        "SEAT",
        "SKODA",
        "SMART",
        "SSANGYONG",
        "SUBARU",
        "SUZUKI",
        "TALBOT",
        "TATA",
        "TESLA",
        "TOYOTA",
        "VOLVO",
        "VW"
    ],
    models: [
        "B-MAX",
        "C-Max",
        "Cougar",
        "Econovan",
        "ECOSPORT",
        "Edge",
        "Escort",
        "Explorer",
        "Fiesta",
        "Fiesta Courier",
        "Focus",
        "Focus C-Max",
        "Fusion",
        "Galaxy",
        "KA",
        "Kuga",
        "Maverick",
        "Mondeo",
        "Mustang",
        "Orion",
        "Probe",
        "Puma",
        "Scorpio",
        "Sierra",
        "S-Max",
        "Streetka",
        "Tourneo Connect",
        "Tourneo Courier",
        "Transit",
        "Transit Connect",
        "Transit Courier",
        "Transit Custom",
        "Windstar"
    ],
    vehicles: [
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 150,
            "enginePowerKW": 110,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1999
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 200,
            "enginePowerKW": 147,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 125,
            "enginePowerKW": 92,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 140,
            "enginePowerKW": 103,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 49,
            "enginePowerKW": 36,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1084
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 45,
            "enginePowerKW": 33,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 935
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 86,
            "enginePowerKW": 63,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1084
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 86,
            "enginePowerKW": 63,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 120,
            "enginePowerKW": 88,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 125,
            "enginePowerKW": 92,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 65,
            "enginePowerKW": 48,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1271
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 182,
            "enginePowerKW": 134,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 135,
            "enginePowerKW": 99,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 150,
            "enginePowerKW": 110,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1999
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1271
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 49,
            "enginePowerKW": 36,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 82,
            "enginePowerKW": 60,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 120,
            "enginePowerKW": 88,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 140,
            "enginePowerKW": 103,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 45,
            "enginePowerKW": 33,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 935
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Gas",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        }
    ]
}

export default defaultState;