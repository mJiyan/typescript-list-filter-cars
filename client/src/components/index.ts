import Header from './Header';
import Card from './Card';
import Filter from './Filter';
import Search from './Search';
import MakeList from './MakeList';
import ModelList from './ModelList';
import CardList from './CardList';


export { Header, Card, Filter, Search, MakeList, ModelList, CardList }