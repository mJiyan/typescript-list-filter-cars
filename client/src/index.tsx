import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import store from './Store';
import App from './App';
import './assets/styles/tailwind.css'

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root'),
);
