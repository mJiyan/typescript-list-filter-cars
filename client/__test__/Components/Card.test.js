import React from 'react';
import { Card } from '../../src/components';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Router, Route, Switch } from 'react-router-dom';
import { createMemoryHistory } from 'history'

import defaultState from '../Mock/defaultState';
import toJson from 'enzyme-to-json'; //added this line
import 'jsdom-global/register'; //at the top of file , even  , before importing react

const mockStore = configureMockStore([thunk]);
const store = mockStore(defaultState);
const history = createMemoryHistory()
//make, model, enginePowerPS, enginePowerKW, fuelType, bodyType, engineCapacity
const wrapper = renderer.create(
    <Provider store={store}>
        <Router history={history}>
            <Switch>
                <Card make={defaultState.vehicles[0].make} model={defaultState.vehicles[0].model} enginePowerPS={defaultState.vehicles[0].enginePowerPS} enginePowerKW={defaultState.vehicles[0].enginePowerKW} fuelType={defaultState.vehicles[0].fuelType} bodyType={defaultState.vehicles[0].bodyType} engineCapacity={defaultState.vehicles[0].engineCapacity} />
            </Switch>

        </Router>

    </Provider>,
);


describe('rendering card components', () => {
    test('it should match with the snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

});