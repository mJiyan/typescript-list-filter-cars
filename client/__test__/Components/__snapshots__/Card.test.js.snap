// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering card components it should match with the snapshot 1`] = `
<div
  className="rounded-md overflow-hidden shadow-lg bg-WhiteDark dark:bg-DarkBlue h-96 w-80 cursor-pointer
                xl:mb-20 lg:mb-20 md:mb-20 sm:mb-10 xl:mb-10"
>
  <img
    className="object-cover w-full h-44"
    src="https://cdn.dribbble.com/users/1111447/screenshots/7697273/rent_a_car_logo.png"
  />
  <div
    className="flex flex-col items-start justify-start h-full p-8"
  >
    <p
      className="text-orange-600 text-lg pb-4 font-bold"
    >
       
      FORD Fiesta
    </p>
    <div
      className="flex flex-col items-start justify-end"
    >
      <p
        className="text-base"
      >
        <span
          className="font-normal"
        >
          enginePowerPS: 
        </span>
         
        60
      </p>
      <p
        className="text-base"
      >
        <span
          className="font-normal"
        >
          enginePowerKW: 
        </span>
         
        44
      </p>
      <p
        className="text-base"
      >
        <span
          className="font-normal"
        >
          fuelType: 
        </span>
         
        Benzin
      </p>
      <p
        className="text-base"
      >
        <span
          className="font-normal"
        >
          bodyType: 
        </span>
         
        Limousine
      </p>
      <p
        className="text-base"
      >
        <span
          className="font-normal"
        >
          engineCapacity: 
        </span>
         
        1299
      </p>
    </div>
  </div>
</div>
`;
